pkg/restore/util.go

```diff
@@ -353,28 +353,44 @@ func truncateTS(key []byte) []byte {
 	return key[:len(key)-8]
 }
 
-// SplitRanges splits region by
+// SplitRangesAndThen splits region by
 // 1. data range after rewrite.
 // 2. rewrite rules.
-func SplitRanges(
+func SplitRangesAndThen(
 	ctx context.Context,
 	client *Client,
 	ranges []rtree.Range,
 	rewriteRules *RewriteRules,
 	updateCh glue.Progress,
-) error {
+	callback func(error),
+) {
 	start := time.Now()
 	defer func() {
 		elapsed := time.Since(start)
 		summary.CollectDuration("split region", elapsed)
 	}()
 	splitter := NewRegionSplitter(NewSplitClient(client.GetPDClient(), client.GetTLSConfig()))
 
-	return splitter.Split(ctx, ranges, rewriteRules, func(keys [][]byte) {
+	callback(splitter.Split(ctx, ranges, rewriteRules, func(keys [][]byte) {
 		for range keys {
 			updateCh.Inc()
 		}
+	}))
+}
+
+// SplitRanges is the synchronous version of SplitRangesAndThen.
+func SplitRanges(
+	ctx context.Context,
+	client *Client,
+	ranges []rtree.Range,
+	rewriteRules *RewriteRules,
+	updateCh glue.Progress,
+) error {
+	ch := make(chan error)
+	go SplitRangesAndThen(ctx, client, ranges, rewriteRules, updateCh, func(err error) {
```

Comment:
Why not 
```
var ret error
SplitRangesAndThen(ctx, client, ranges, rewriteRules, updateCh, func(err error) {
     ret = err
})
return ret
```

Reply:
Even I don't know why, this ruins the whole procedure.

Thankfully, after using `errgroup`, we don't need this change anymore. 😄